### PR TITLE
samples: fix flaky test teardown in servicedirectory

### LINKED ISSUE #6972: servicedirectory.snippets_test: test_delete_namespace failed
This test failed!

To configure my behavior, see [the Flaky Bot documentation](https://github.com/googleapis/repo-automation-bots/tree/main/packages/flakybot).

If I'm commenting on this issue too often, add the `flakybot: quiet` label and
I will stop commenting.

---

commit: 34bdda252cbbe0f0b33439790a6bbc95be6150cc
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/5fe50c94-b53d-400d-aae4-12129b28a994), [Sponge](http://sponge2/5fe50c94-b53d-400d-aae4-12129b28a994)
status: failed
<details><summary>Test output</summary><br><pre>Traceback (most recent call last):
  File "/workspace/servicedirectory/.nox/py-3-6/lib/python3.6/site-packages/google/api_core/grpc_helpers.py", line 66, in error_remapped_callable
    return callable_(*args, **kwargs)
  File "/workspace/servicedirectory/.nox/py-3-6/lib/python3.6/site-packages/grpc/_channel.py", line 946, in __call__
    return _end_unary_response_blocking(state, call, False, None)
  File "/workspace/servicedirectory/.nox/py-3-6/lib/python3.6/site-packages/grpc/_channel.py", line 849, in _end_unary_response_blocking
    raise _InactiveRpcError(state)
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.NOT_FOUND
	details = "Resource 'projects/python-docs-samples-tests/locations/us-east1/namespaces/test-namespace-1912544e8fdc410faa35ef8f7e24e702' does not exist."
	debug_error_string = "{"created":"@1635327013.693113122","description":"Error received from peer ipv4:74.125.195.95:443","file":"src/core/lib/surface/call.cc","file_line":1069,"grpc_message":"Resource 'projects/python-docs-samples-tests/locations/us-east1/namespaces/test-namespace-1912544e8fdc410faa35ef8f7e24e702' does not exist.","grpc_status":5}"
>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/workspace/servicedirectory/snippets_test.py", line 38, in teardown_module
    namespace = client.get_namespace(name=namespace_name)
  File "/workspace/servicedirectory/.nox/py-3-6/lib/python3.6/site-packages/google/cloud/servicedirectory_v1/services/registration_service/client.py", line 663, in get_namespace
    response = rpc(request, retry=retry, timeout=timeout, metadata=metadata,)
  File "/workspace/servicedirectory/.nox/py-3-6/lib/python3.6/site-packages/google/api_core/gapic_v1/method.py", line 142, in __call__
    return wrapped_func(*args, **kwargs)
  File "/workspace/servicedirectory/.nox/py-3-6/lib/python3.6/site-packages/google/api_core/grpc_helpers.py", line 68, in error_remapped_callable
    raise exceptions.from_grpc_error(exc) from exc
  File "/workspace/servicedirectory/.nox/py-3-6/lib/python3.6/site-packages/google/api_core/exceptions.py", line 532, in from_grpc_error
    if isinstance(rpc_exc, grpc.Call) or _is_informative_grpc_error(rpc_exc):
AttributeError: 'NoneType' object has no attribute 'Call'</pre></details>

### PR DESCRIPTION
**Pull request by parthea**
Fixes #6972

### CODE REVIEW

**Review by busunkim96 — COMMENTED** [review]
Could you explain what the source of the flakiness was?

File: servicedirectory/snippets_test.py [hunk 1]
```diff
@@ -34,11 +35,16 @@
...
-        pass
+    all_namespaces = quickstart.list_namespaces(PROJECT_ID, LOCATION_ID).namespaces
+
+    # Delete namespace only if it exists
+    for namespace in all_namespaces:
+        if namespace_name in namespace.name:
+            try:
+                client.delete_namespace(name=namespace_name)
+            except exceptions.NotFound:
+                pass
```
Comment (leahecole) [inline servicedirectory/snippets_test.py:46]:
```suggestion
                print("Namespace already deleted")
```

Comment (leahecole) [reply servicedirectory/snippets_test.py:46]: just a small nit - except/pass is a pet peeve of mine and I like to print something useful instead

**Review by leahecole — CHANGES_REQUESTED** [review]

File: servicedirectory/snippets_test.py (hunk 1 above)
Comment (parthea) [reply servicedirectory/snippets_test.py]: I agree! Done.

**Comment by parthea** [comment]
@busunkim96 Please could you review/approve?

**Review by leahecole — COMMENTED** [review]
i'm happy if @busunkim96 is :)

**Review by busunkim96 — APPROVED** [review]

**Comment by busunkim96** [comment]
Thank you @parthea!

**Review by leahecole — APPROVED** [review]

**Pull request merged**